lms/util/lti_launch.py

```diff
@@ -11,55 +11,46 @@ def get_application_instance(session, consumer_key):
     ).one()
 
 
-def default_get_secret(request, consumer_key):
+def get_secret(request, consumer_key):
     """Retrieve the lti secret given."""
     instance = get_application_instance(request.db, consumer_key)
     return instance.shared_secret
 
 
-def default_get_lti_launch_params(request):
+def get_lti_launch_params(request):
     """Retrieve the lti launch params."""
     return dict(request.params)
 
 
-def lti_launch(get_lti_launch_params=default_get_lti_launch_params,
-               get_secret=default_get_secret):
+def lti_launch(view_function):
     """
-    Initialize decorator.
+    Handle the verification of an lms launch.
 
-    Allow caller to supply methods to customize extracting lti launch
-    params and retrieving lti secret
-    """
-    def decorator(view_function):
-        """
-        Handle the verification of an lms launch.
-
-        You should add this decorator before (logically) the route decorator.
-        For example:
-
-        @view_config(...)
-        @lti_launch
-        def some_view(request):
-        ...
-        """
-        def wrapper(request):
-            """Handle the lms validation."""
-            lti_params = get_lti_launch_params(request)
-            consumer_key = lti_params['oauth_consumer_key']
-            shared_secret = get_secret(request, consumer_key)
+    You should add this decorator before (logically) the route decorator.
+    For example:
 
-            consumers = {}
-
-            consumers[consumer_key] = {"secret": shared_secret}
-            # TODO rescue from an invalid lms launch
-            pylti.common.verify_request_common(
-                consumers,
-                request.url,
-                request.method,
-                dict(request.headers),
-                dict(lti_params))
-            jwt_secret = request.registry.settings['jwt_secret']
-            jwt_token = build_jwt_from_lti_launch(request.params, jwt_secret)
-            return view_function(request, jwt_token)
-        return wrapper
-    return decorator
+    @view_config(...)
+    @lti_launch
+    def some_view(request):
+    ...
+    """
+    def wrapper(request):
+        """Handle the lms validation."""
+        lti_params = get_lti_launch_params(request)
+        consumer_key = lti_params['oauth_consumer_key']
```

Comment:
I presume that these areas are where @sheetaluk is working on errors/raising Exceptions?

Reply:
I guess so! I suspect there are many, many ways for the app to crash on missing or invalid params, we haven't found them all yet, and fixing some of them will simply reveal more. The whack-a-mole approach to fixing it probably isn't the most efficient but then again getting something like validation schemas in place might require non-simple refactoring